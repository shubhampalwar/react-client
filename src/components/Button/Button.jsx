import React, { Component } from 'react';
import PropTypes from 'prop-types';
import btnStyle from './style';

class Button extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const {
      color, disabled, style, ...rest
    } = this.props;
    const btnColor = !btnStyle[color] || color === 'default' || disabled
      ? {}
      : btnStyle[color];
    return (
      <>
        <input id="Button" style={{ ...btnStyle.base, ...btnColor }} disabled={disabled} type="button" {...rest} />
      </>
    );
  }
}

Button.propTypes = {
  color: PropTypes.string,
  disabled: PropTypes.bool,
  style: PropTypes.objectOf(PropTypes.string),
  value: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
Button.defaultProps = {
  disabled: false,
  color: 'primary',
  style: {},
};
export default Button;
