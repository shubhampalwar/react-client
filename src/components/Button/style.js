export default {
  base: {
    padding: '12px',
    margin: '0px 5px 0px 0px',
    borderRadius: '5px',
    border: '1px solid lightgrey',
    boxSizing: 'border-box',
    fontWeight: 'bold',
    cursor: 'pointer',
  },
  primary: {
    border: '#4caf50',
    backgroundColor: '#4caf50',
    color: 'white',
    cursor: 'pointer',
  },
};
