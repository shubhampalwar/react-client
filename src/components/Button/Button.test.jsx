import { shallow } from 'enzyme';
import React from 'react';
import Button from './Button';

const wrapper = shallow(<Button />);

describe('Button', () => {
  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
