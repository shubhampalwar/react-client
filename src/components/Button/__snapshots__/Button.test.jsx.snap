// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    color="primary"
    disabled={false}
    style={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <input
        disabled={false}
        id="Button"
        style={
          Object {
            "backgroundColor": "#4caf50",
            "border": "#4caf50",
            "borderRadius": "5px",
            "boxSizing": "border-box",
            "color": "white",
            "cursor": "pointer",
            "fontWeight": "bold",
            "margin": "0px 5px 0px 0px",
            "padding": "12px",
          }
        }
        type="button"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "disabled": false,
        "id": "Button",
        "style": Object {
          "backgroundColor": "#4caf50",
          "border": "#4caf50",
          "borderRadius": "5px",
          "boxSizing": "border-box",
          "color": "white",
          "cursor": "pointer",
          "fontWeight": "bold",
          "margin": "0px 5px 0px 0px",
          "padding": "12px",
        },
        "type": "button",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <input
          disabled={false}
          id="Button"
          style={
            Object {
              "backgroundColor": "#4caf50",
              "border": "#4caf50",
              "borderRadius": "5px",
              "boxSizing": "border-box",
              "color": "white",
              "cursor": "pointer",
              "fontWeight": "bold",
              "margin": "0px 5px 0px 0px",
              "padding": "12px",
            }
          }
          type="button"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "disabled": false,
          "id": "Button",
          "style": Object {
            "backgroundColor": "#4caf50",
            "border": "#4caf50",
            "borderRadius": "5px",
            "boxSizing": "border-box",
            "color": "white",
            "cursor": "pointer",
            "fontWeight": "bold",
            "margin": "0px 5px 0px 0px",
            "padding": "12px",
          },
          "type": "button",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
